File: java_patterns/src/Basic_Patterns/Behavioural_Patterns/Visitor/Concept/code/Visitors/Visitor.java
package Basic_Patterns.Behavioural_Patterns.Visitor.Concept.code.Visitors;

import Basic_Patterns.Behavioural_Patterns.Visitor.Concept.code.Elements.ConcreteElementA;
import Basic_Patterns.Behavioural_Patterns.Visitor.Concept.code.Elements.ConcreteElementB;

public interface Visitor {
    void visit(ConcreteElementA element);
    void visit(ConcreteElementB element);
}

File: src/Basic_Patterns/Behavioural_Patterns/Visitor/Concept/code/visitors/Visitor.ts
import {ConcreteElementB} from "src/Basic_Patterns/Behavioural_Patterns/Visitor/Concept/code/elements/ConcreteElementB";
import {ConcreteElementA} from "src/Basic_Patterns/Behavioural_Patterns/Visitor/Concept/code/elements/ConcreteElementA";

interface Visitor {
    visitConcreteElementA(element: ConcreteElementA): void;
    visitConcreteElementB(element: ConcreteElementB): void;
}

export { Visitor };